sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py

```diff
@@ -913,119 +1109,319 @@ async def execute_tool_calls(self, tool_calls: List[Any]) -> Any:
         return tool_outputs
 
 
-class AgentEventHandler:
+CustomDataT = TypeVar("CustomDataT")
+T = TypeVar("T")
+BaseAsyncAgentEventHandlerT = TypeVar("BaseAsyncAgentEventHandlerT", bound="BaseAsyncAgentEventHandler")
+BaseAgentEventHandlerT = TypeVar("BaseAgentEventHandlerT", bound="BaseAgentEventHandler")
 
-    def on_message_delta(self, delta: "MessageDeltaChunk") -> None:
+
+class BaseAsyncAgentEventHandler(AsyncIterator[T]):
+    done = False
```

Comment:
Why do we have class members attributes?

Reply:
These class member attributes are the most essential attributes used by the base class.
When users sub-class BaseAsyncAgentEventHandler, they might write their own constructor and not instantiate through our default constructor.   So I don't set them in constructor.

Reply:
That would be a bug/programming error in the derived class. It is a slippery slope to be forgiving w. regards to covering user errors.

Reply:
What you said is right?  But I always put the best effort to minimize issue caused by their errors.   Do you need me move these code?

Reply:
Fixed.